### PR TITLE
Slime Crossbreeding - Chilling Extracts!

### PR DESCRIPTION
**Pull request by fludd12**
:cl:fludd12
add: Chilling extracts are now available through crossbreeding. Obtain them by feeding 10 dark blue extracts to an adult slime! Their full list of effects is available on the wiki.
/:cl:

Is it just me, or is it getting cold in here?
**Sprites are here!**

### CODE REVIEW

**Comment by Borisvanmemes** [comment]
Oh god yes

**Comment by fludd12** [comment]
We're halfway done.

...Saying that hurts me, actually. :scream_cat:

**Comment by Tlaltecuhtli** [comment]
`>`check wiki for the extracts 
`>`nothing there

**Comment by fludd12** [comment]
I'll put them there later.

Besides, that's for the PLAYERS, dork :>

**Comment by fludd12** [comment]
ohjeez i forgot a few sprites one moment

**Comment by fludd12** [comment]
@Tlaltecuhtli Wiki updated.

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (Tlaltecuhtli) [inline code/modules/research/xenobiology/crossbreeding/chilling.dm]: cant you make it check for factions between the mob and the user instead? xenobio can spawn mobs with the slime faction

Comment (fludd12) [reply code/modules/research/xenobiology/crossbreeding/chilling.dm]: I could, but because regular gold cores only spawn with the neutral faction, it may be a bit restrictive.

**Comment by Dorsisdwarf** [comment]
I feel like this PR should mention whatever the behind-the-scenes stuff you're doing with conflating bodyparts and organs?

**Comment by fludd12** [comment]
Preparation for the next PR. I forgot to remove it when copying over my old branch. I can remove it and add it in the next PR, but there's not really a need to - basically, what it's doing is adding the defines for hand and foot organ slots, for implants / replacement hands.

As for the component signalling in ranged attack and throwing, that'll also be useful later, but I figured they're pretty nice to have now, if anyone wants them.

I can atomize better if you'd like!

**Comment by fludd12** [comment]
Realization: I forgot to include status effect sprite updates.

Wishing I had written down all the changes I made in the old PR :/

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (Tlaltecuhtli) [reply code/modules/research/xenobiology/crossbreeding/chilling.dm]: luminol race + gold extract + harm intent makes 1 mob with slime faction there may be other edge cases also slime pokeball

**Comment by fludd12** [comment]
...Is travis hung up on something?

File: code/__DEFINES/DNA.dm
Comment (ninjanomnom) [inline code/__DEFINES/DNA.dm]: These don't appear to be used anywhere? what are they doing in here anyway

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/__DEFINES/DNA.dm
Comment (fludd12) [reply code/__DEFINES/DNA.dm]: aaaa i literally mentioned that earlier i can remove if you'd like but there are plans for them

Comment (fludd12) [reply code/__DEFINES/DNA.dm]: no one said anything to me about them for four days

Comment (ninjanomnom) [reply code/__DEFINES/DNA.dm]: If you want a timely answer and there isn't obviously a maintainer keeping an eye on the pr you should get on the irc and ask one of us there.

In short though you should remove this if you're not using it here.

File: code/_onclick/other_mobs.dm
Comment (ninjanomnom) [inline code/_onclick/other_mobs.dm]: Make sure to update your branch to latest, this method of calling on signals is outdated now.

File: code/modules/mob/living/carbon/carbon.dm
Comment (ninjanomnom) [inline code/modules/mob/living/carbon/carbon.dm]: This signal should be much earlier in the proc

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/_onclick/other_mobs.dm
Comment (fludd12) [reply code/_onclick/other_mobs.dm]: shooooooot

File: code/modules/research/xenobiology/crossbreeding/_status_effects.dm
Comment (ninjanomnom) [inline code/modules/research/xenobiology/crossbreeding/_status_effects.dm]: I'm absolutely horrified this was let through earlier, half assing the conversion of status effects to components is terrible.

Comment (fludd12) [reply code/modules/research/xenobiology/crossbreeding/_status_effects.dm]: ...aw.
i put a lot of work into that.
any suggestions?

Comment (ninjanomnom) [reply code/modules/research/xenobiology/crossbreeding/_status_effects.dm]: It's not appropriate in this pr but the status effect type path should have been removed entirely

File: code/modules/mob/living/carbon/carbon.dm
Comment (ninjanomnom) [reply code/modules/mob/living/carbon/carbon.dm]: It's standard for us to have these sorts of signals early on in the proc so that components which prevent it do so before it has to do other checks that are now unnecesary.

Do you even use this in this pr?

Comment (fludd12) [reply code/modules/mob/living/carbon/carbon.dm]: ...Huh, apparently not. I thought I did. Will add in relevant PR later.

Also, I think I'll refactor all status effects, then. Component status effects would allow for non-living status effects, too, such as making burning items a status effect, et cetera.

File: code/modules/research/xenobiology/crossbreeding/_status_effects.dm
Comment (ninjanomnom) [reply code/modules/research/xenobiology/crossbreeding/_status_effects.dm]: With #38537 removing this, there are heavy conflicts between the two prs. I'll leave it up to @Cyberboss whether he wants to fix the conflicts or make @fludd12 handle it depending on merge order.

**Comment by fludd12** [comment]
darn you tgstation server

File: code/modules/research/xenobiology/crossbreeding/_status_effects.dm
Comment (fludd12) [reply code/modules/research/xenobiology/crossbreeding/_status_effects.dm]: I'll use cyberboss' method - no sense halting another PR because of me.

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (ninjanomnom) [inline code/modules/research/xenobiology/crossbreeding/chilling.dm]: Split these into other files.

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (ninjanomnom) [inline code/modules/research/xenobiology/crossbreeding/chilling.dm]: `. = ..()`

why is it after?

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (ninjanomnom) [inline code/modules/research/xenobiology/crossbreeding/chilling.dm]: `return ..()`

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (ninjanomnom) [inline code/modules/research/xenobiology/crossbreeding/chilling.dm]: `TRUE`

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (ninjanomnom) [inline code/modules/research/xenobiology/crossbreeding/chilling.dm]: what

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (fludd12) [reply code/modules/research/xenobiology/crossbreeding/chilling.dm]: :tangerine:

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (fludd12) [reply code/modules/research/xenobiology/crossbreeding/chilling.dm]: :tangerine:

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (fludd12) [reply code/modules/research/xenobiology/crossbreeding/chilling.dm]: I'm

I'm not sure what I was thinking at the time.

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (fludd12) [reply code/modules/research/xenobiology/crossbreeding/chilling.dm]: Will do, especially since lots of other files have this kinda thing.

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (fludd12) [reply code/modules/research/xenobiology/crossbreeding/chilling.dm]: This was actually intentional, the wall is intended to be wholly impenetrable but brief

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (fludd12) [reply code/modules/research/xenobiology/crossbreeding/chilling.dm]: Done!

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (fludd12) [reply code/modules/research/xenobiology/crossbreeding/chilling.dm]: Suggestions to go about that...?

Comment (ninjanomnom) [reply code/modules/research/xenobiology/crossbreeding/chilling.dm]: Just count how much plasma you're removing and stop early if you go over a certain amount

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (ninjanomnom) [reply code/modules/research/xenobiology/crossbreeding/chilling.dm]: You didn't read what detect_room() does did you

File: code/modules/research/xenobiology/crossbreeding/_products.dm
Comment (ninjanomnom) [inline code/modules/research/xenobiology/crossbreeding/_products.dm]: This applies to the whole file, split things up some more. Clothing in the clothing module etc

File: code/modules/research/xenobiology/crossbreeding/_products.dm
Comment (ninjanomnom) [inline code/modules/research/xenobiology/crossbreeding/_products.dm]: You're not processing any more

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (ninjanomnom) [inline code/modules/research/xenobiology/crossbreeding/chilling.dm]: This line is unnecessary now

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (ninjanomnom) [inline code/modules/research/xenobiology/crossbreeding/chilling.dm]: You still need to deal with the issue of hotspots getting created on the other side of walls.

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by fludd12** [comment]
Sorry for the delays in correcting things, have been exhausted from work. Will hopefully fix everything today.

File: code/modules/research/xenobiology/crossbreeding/_products.dm
Comment (fludd12) [reply code/modules/research/xenobiology/crossbreeding/_products.dm]: Hegh, that's gonna be a lot of reorganizing.

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (fludd12) [reply code/modules/research/xenobiology/crossbreeding/chilling.dm]: Er, not to question too much, but why would that be a *bad* thing...? Phlogiston and such does that.

Comment (ninjanomnom) [reply code/modules/research/xenobiology/crossbreeding/chilling.dm]: A lot of ss13's code is terrible for various reasons, doesn't mean you have to follow suit.

**Comment by AnturK** [comment]
So how is this going ?

**Comment by fludd12** [comment]
Tearfully

File: code/modules/research/xenobiology/crossbreeding/chilling.dm
Comment (ninjanomnom) [reply code/modules/research/xenobiology/crossbreeding/chilling.dm]: Probably the simplest and most interesting way to solve the issue would be to make it shoot actual fire projectiles

Comment (fludd12) [reply code/modules/research/xenobiology/crossbreeding/chilling.dm]: Ooh, good plan!

**Comment by Cyberboss** [comment]
You're PR is getting old. Either Travis CI is failing, it is still labelled work in progress (change this by adding `[READY]` to the start of your title), or there are unaddressed change requests. This may be closed soon to free up space on our tracker if it remains idle. In the event that it is, please ask in #coderbus on Rizon for it to be reopened once you have resumed work on it.

**Comment by ExcessiveUseOfCobblestone** [comment]
@Cyberboss get the bot to autogenerate those messages lol

**Comment by Cyberboss** [comment]
Github has saved replies and i;m too lazy to cod eit

**Comment by fludd12** [comment]
I'll do this I *swear* I'm so sorry

File: code/modules/research/xenobiology/crossbreeding/_products.dm
Comment (fludd12) [reply code/modules/research/xenobiology/crossbreeding/_products.dm]: I think I'll end up doing a separate PR to reorganize everything into specific modules, if that's okay with you? It really will be a *ton* of reorganizing, and I want to get it out of the way before continuing with anything else.

File: code/modules/research/xenobiology/crossbreeding/_products.dm
Comment (fludd12) [reply code/modules/research/xenobiology/crossbreeding/_products.dm]: Right!

File: code/modules/research/xenobiology/crossbreeding/_products.dm
Comment (ninjanomnom) [reply code/modules/research/xenobiology/crossbreeding/_products.dm]: Yeah that's a no go. You need to think about the thing you're coding and put it into the appropriate place.

**Comment by optimumtact** [comment]
There is no rush, make a new PR when you feel ready and comfortable, don't feel too pushed to meet others expectations

**Pull request closed without merging**
